olp-cpp-sdk-core/src/client/OlpClientSettingsFactory.cpp

```diff
@@ -42,8 +45,12 @@ OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler(
 std::unique_ptr<cache::KeyValueCache>
 OlpClientSettingsFactory::CreateDefaultCache(cache::CacheSettings settings) {
   auto cache = std::make_unique<cache::DefaultCache>(std::move(settings));
-  cache->Open();
-  return std::move(cache);
+  auto error = cache->Open();
+  if (error == cache::DefaultCache::StorageOpenResult::OpenDiskPathFailure)
+  {
+      OLP_SDK_LOG_FATAL(kLogTag, "Error opening disk cache");
+  }
+  return nullptr;
```

Comment:
In case of an open error you should return nullptr only! In case open is successfully you return as before the cache instance.